src/state_point.F90

```diff
@@ -866,14 +790,12 @@ subroutine load_state_point()
       ! Read in sum and sum squared
       if (int_array(1) == 1) then
         TALLY_RESULTS: do i = 1, n_tallies
-
           ! Set pointer to tally
           tally => tallies(i)
-          curr_key = key_array(id_array(i))
 
           ! Read sum and sum_sq for each bin
           tally_group = open_group(tallies_group, "tally " // &
-               trim(to_str(curr_key)))
```

Comment:
There are probably many things that could go wrong in the hands of a user who changes their input and then tries to restart. I think we need should decide on an overall strategy for what we allow to be changed in a restart run before trying to catch errors.